plugins/output/flight/flight.go

```diff
@@ -131,11 +99,124 @@ func (f *Flight) Connect() error {
 	return nil
 }
 
+// Write should write immediately to the output, and not buffer writes
+// (Telegraf manages the buffer for you). Returning an error will fail this
+// batch of writes and the entire batch will be retried automatically.
+func (f *Flight) Write(metrics []telegraf.Metric) error {
+
+	// Create a new RecordBuilder using the schema.
+	builder := array.NewRecordBuilder(memory.DefaultAllocator, &f.schema)
+	defer builder.Release()
+
+	schemaFields := f.schema.Fields()
+
+	builder.Reserve(len(metrics))
+
+	// Iterate through the metrics and add them to the RecordBuilder.
+	for _, metric := range metrics {
+
+		addMetricToRecordBuilder(schemaFields, builder, metric)
+
+	}
+
+	// Create a new Record from the RecordBuilder.
+	rec := builder.NewRecord()
+	defer rec.Release()
+
+	return f.writer.Write(rec)
+}
+
 // Close the connection to the Apache Arrow Flight server.
 func (f *Flight) Close() error {
 	return f.writer.Close()
 }
 
-func init() {
-	outputs.Add("flight", func() telegraf.Output { return &Flight{} })
+// Iterate through the fields in the schema,
+// extract the tag or field values from the metric using getTag or getField,
+// and add the corresponding value to the correct builder in RecordBuilder.
+// If the value is not set in the metric, panic and print the error.
+//
+// The switches for the types inside each case are necessary because the type
+// represented by the schema is not always the same as the type represented by the metric.
+// For example, JSON does not define types and therefore parses all integers
+// as floats. If the schema expects an integer in the field, Apache Arrow will panic.
+func addMetricToRecordBuilder(schemaFields []arrow.Field, builder *array.RecordBuilder, metric telegraf.Metric) {
+
+	timeInt := metric.Time().UnixMilli()
+
+	for i, schemaField := range schemaFields {
+		switch schemaField.Type.ID() {
+		case arrow.TIMESTAMP:
+			builder.Field(i).(*array.TimestampBuilder).AppendValues([]arrow.Timestamp{arrow.Timestamp(timeInt)}, nil)
```

Comment:
Can [`.AppendValues()`](https://pkg.go.dev/github.com/apache/arrow/go/arrow/array#TimestampBuilder.AppendValues) be replaced with[ `.Append()`](https://pkg.go.dev/github.com/apache/arrow/go/arrow/array#TimestampBuilder.Append) everywhere to not create a slice for each element that is added to the builders?

Reply:
Fixed.